modules/vaos/app/services/eps/appointment_status_notification_callback.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+module Eps
+  # Custom callback class for handling VA Notify email delivery status updates
+  # for appointment status notifications.
+  #
+  # This callback tracks delivery success/failure and provides observability
+  # through StatsD metrics and structured logging.
+  class AppointmentStatusNotificationCallback
+    STATSD_KEY = 'api.vaos.appointment_status_notification'
+    FAILURE_STATUSES = %w[permanent-failure temporary-failure technical-failure].freeze
+    FAILURE_TYPE_MAP = {
+      'permanent-failure' => 'permanent',
+      'temporary-failure' => 'temporary',
+      'technical-failure' => 'technical'
+    }.freeze
+
+    # Main callback entry point called by VA Notify
+    #
+    # @param notification [VANotify::Notification] The notification object from VA Notify
+    # @return [void]
+    def self.call(notification)
+      return handle_missing_notification if notification.nil?
+
+      metadata = extract_metadata(notification)
+      base_data = build_base_data(notification, metadata)
+
+      handle_notification_status(notification, base_data)
+    rescue => e
+      handle_callback_error(e, notification)
+    end
+
+    # Extract and validate callback metadata
+    #
+    # @param notification [VANotify::Notification] The notification object
+    # @return [Hash] The callback metadata hash
+    def self.extract_metadata(notification)
+      metadata = notification.callback_metadata || {}
+
+      # Log warning if metadata is missing or incomplete
+      if metadata.empty? || metadata['user_uuid'].blank? || metadata['appointment_id_last4'].blank?
+        Rails.logger.warn(
+          'Community Care Appointments: Eps::AppointmentNotificationCallback received missing or incomplete metadata',
+          notification_id: notification.notification_id,
+          metadata_present: !metadata.empty?,
+          user_uuid_present: metadata['user_uuid'].present?,
+          appointment_id_present: metadata['appointment_id_last4'].present?,
+          status: notification.status
+        )
+        StatsD.increment(
+          "#{STATSD_KEY}.missing_metadata",
+          tags: [
+            'Community Care Appointments',
+            "user_uuid:#{metadata['user_uuid']}",
+            "appointment_id_last4:#{metadata['appointment_id_last4']}"
+          ]
```

Comment:
Isn't the user_uuid and appointment_id going to cause high cardinality here? Using unique tags in Datadog drives up costs, so we're working to keep them to a minimum

Reply:
@LindseySaari I didn't realize this was the case, thank you for bringing it to my attention, updated 👍🏼